### PR TITLE
[Synapse] `az synapse workspace create/update`: Support workspace encryption and user-assignment management identity

### PR DESCRIPTION
**Pull request by kevinzz6**
**Related command**

**Description**

Add new parameters to support synapse workspace encrytion function and user-assigment management identity function.
**Testing Guide**

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Synapse

**Comment by jsntcy** [comment]
@kevinzz6, please fix CI error first

**Comment by kevinzz6** [comment]
@jsntcy please take a time to review it, thanks

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py [hunk 1]
```diff
@@ -102,6 +102,9 @@ def load_arguments(self, _):
             c.argument('project_name', arg_group=repository_arg_group, help='The project name to which you are connecting.')
             c.argument('tenant_id', arg_group=repository_arg_group, help='The tenant id used to connect Azure devops')
             c.argument('last_commit_id', arg_group=repository_arg_group, help='The last commit ID.')
+            c.argument('user_assigned_identity_id', options_list=['--uami-id'], nargs='+', help='The list of User-assigned Managed Identity Id for workspace.')
+            c.argument('user_assigned_identity_in_encryption', options_list=['--uami-id-in-encrypt'], help='User assigned identity resource Id used in Workspace Encryption')
+            c.argument('use_system_assigned_identity_in_encryption', options_list=['--use-sami-in-encrypt'], help='Whether use System assigned identity in Workspace Encryption. If use uami, please set True.If not, set False')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:107]: Is this parameter string type or bool type?

Comment (kevinzz6) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:107]: string type.

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py (hunk 1 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:107]: Do you want to restrict the input within `['True', 'False', 'true', 'false']`?

Comment (kevinzz6) [reply src/azure-cli/azure/cli/command_modules/synapse/manual/_params.py:107]: No, I don't want that.

**Pull request merged**